wp-admin/js/widgets/media-widgets.js

```diff
@@ -726,6 +726,12 @@ wp.mediaWidgets = ( function( $ ) {
 				modelProps.height = mediaFrameProps.customHeight;
 			}
 
+			if ( 'post' === mediaFrameProps.link ) {
+				modelProps.link_url = control.selectedAttachment.get( 'link' );
```

Comment:
This feels a bit “impure” in that we're looking at `control.selectedAttachment` but we're specifically being passed `mediaFrameProps`. I don't suppose `mediaFrameProps.link` is possible?